src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -148,14 +149,32 @@ public void Start()
             _initialInstrumentEnumerationComplete();
         }
 
+        public void Update()
+        {
+            var publishedInstruments = Meter.GetPublishedInstruments();
+
+            if (publishedInstruments is not null)
+            {
+                foreach (Instrument instrument in publishedInstruments)
+                {
+                    if (!_instrumentList.Contains(instrument))
+                    {
+                        _listener.InstrumentPublished?.Invoke(instrument, _listener);
```

Comment:
This works, but I think it is misleading. I would refactor the handler for this event into a named method on AggregationManager rather than a lambda and then use that method both here and in the callback registration above.